core/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -1257,12 +1260,22 @@ public boolean equals(Object obj) {
 
         @Override
         public int hashCode() {
-            return 31 * 31 * 31 + 31 * 31 * Long.hashCode(seqNo) + 31 * Long.hashCode(primaryTerm) + reason().hashCode();
+            int result = (int) (seqNo ^ (seqNo >>> 32));
+            result = 31 * result + (int) (primaryTerm ^ (primaryTerm >>> 32));
+            result = 31 * result + reason.hashCode();
```

Comment:
I'm just going to roll this back. It's not related to my changed and it seems it's contentious.